src/HavenoDaemon.test.ts

```diff
@@ -922,6 +922,225 @@ test("Can complete a trade", async () => {
+  for (let trade of trades) {
+    expect(trade.getPhase()).toEqual("DEPOSIT_PUBLISHED");
+    let fetchedTrade: TradeInfo = await bob.getTrade(trade.getTradeId());
+    expect(fetchedTrade.getPhase()).toEqual("DEPOSIT_PUBLISHED");
+    depositTxIds.push(fetchedTrade.getMakerDepositTxId());
+    depositTxIds.push(fetchedTrade.getTakerDepositTxId());
+  }
+  
+  // mine until deposit txs unlock
+  HavenoUtils.log(1, "Mining to unlock deposit txs");
+  await waitForUnlockedTxs(...depositTxIds);
+  HavenoUtils.log(1, "Done mining to unlock deposit txs");
+  
+  // open disputes
+  HavenoUtils.log(1, "Opening disputes");
+  await bob.openDispute(trades[0].getTradeId());
+  await alice.openDispute(trades[1].getTradeId());
+  await bob.openDispute(trades[2].getTradeId());
+  await alice.openDispute(trades[3].getTradeId());
+  
+  // test dispute
+  let bobDispute = await bob.getDispute(trades[0].getTradeId());
+  expect(bobDispute.getTradeId()).toEqual(trades[0].getTradeId());
+  expect(bobDispute.getIsOpener()).toBe(true);
+  expect(bobDispute.getDisputeOpenerIsBuyer()).toBe(false);
+  
+  // get non-existing dispute should fail
+  try {
+    await bob.getDispute("invalid");
+    throw new Error("get dispute with invalid id should fail");
+  } catch (err) {
+    assert.equal(err.message, "dispute for trade id 'invalid' not found");
+  }
+  
+  // alice sees the dispute
+  await wait(TestConfig.maxTimePeerNoticeMs * 2);
+  let aliceDispute = await alice.getDispute(trades[0].getTradeId());
+  expect(aliceDispute.getTradeId()).toEqual(trades[0].getTradeId());
+  expect(aliceDispute.getIsOpener()).toBe(false);
+  
+  // arbitrator sees both disputes
+  let disputes = await arbitrator.getDisputes();
+  expect(disputes.length).toBeGreaterThanOrEqual(2);
+  let arbAliceDispute = disputes.find(d => d.getId() === aliceDispute.getId());
+  assert(arbAliceDispute);
+  let arbBobDispute = disputes.find(d => d.getId() === bobDispute.getId());
+  assert(arbBobDispute);
+  
+  // arbitrator sends chat messages to alice and bob
+  HavenoUtils.log(1, "Testing chat messages");
+  await arbitrator.sendDisputeChatMessage(arbBobDispute!.getId(), "Arbitrator chat message to Bob", []);
+  await arbitrator.sendDisputeChatMessage(arbAliceDispute!.getId(), "Arbitrator chat message to Alice", []);
+  
+  // alice and bob reply to arbitrator chat messages
+  let attachment = new Attachment();
+  let bytes = new Uint8Array(Buffer.from("Proof Bob was scammed", "utf8"));
+  attachment.setBytes(bytes);
+  attachment.setFileName("proof.txt");
+  let attachment2 = new Attachment();
+  let bytes2 = new Uint8Array(Buffer.from("picture bytes", "utf8"));
+  attachment2.setBytes(bytes2);
+  attachment2.setFileName("proof.png");
+  await bob.sendDisputeChatMessage(bobDispute.getId(), "Bob chat message", [attachment, attachment2]); 
+  await alice.sendDisputeChatMessage(aliceDispute.getId(), "Alice chat message", []); 
+  await wait(TestConfig.maxTimePeerNoticeMs);
```

Comment:
Let's also move this wait down after line 1028 for consistency. Thanks.

Reply:
done